tests/phpunit/FileStateRegistryTest.php

```diff
@@ -5,84 +5,59 @@
 namespace Keboola\FtpExtractor\Tests;
 
 use Keboola\FtpExtractor\FileStateRegistry;
-use Keboola\Temp\Temp;
 use PHPUnit\Framework\TestCase;
-use Symfony\Component\Filesystem\Filesystem;
 
 class FileStateRegistryTest extends TestCase
 {
-
-    public function testRegistry(): void
+    /**
+     * @dataProvider sameSecondUpdateDataProvider
+     * @dataProvider firstRunDataProvider
+     */
+    public function testRegistry(string $path, int $timestamp, FileStateRegistry $registry, bool $expected): void
     {
-        $temp = new Temp('state-registry');
-        $temp->initRunFolder();
-        $dataDir = $temp->getTmpFolder();
-
-        $this->createStateFile($dataDir . '/in/state.json');
-
-        $registry = new FileStateRegistry($dataDir);
-
-        $this->assertTrue($registry->shouldBeFileUpdated('file-abcd-1234.txt', 1541410003));
-        $registry->saveFileTimestamp('file-abcd-1234.txt', 1541410003);
-        $this->assertFalse($registry->shouldBeFileUpdated('file_defg-4567.bin', 1541410005));
-        $registry->saveFileTimestamp('file_defg-4567.bin', 1541410005);
-        $this->assertFalse($registry->shouldBeFileUpdated('test_4444-4567.bin', 1541410009));
-        $registry->saveFileTimestamp('test_4444-4567.bin', 1541410009);
-
-        $registry->saveState();
+        $this->assertSame($expected, $registry->shouldBeFileUpdated($path, $timestamp));
+    }
 
-        $outState = file_get_contents($dataDir . '/out/state.json');
+    public function firstRunDataProvider(): array
+    {
+        $registry = $this->getRegistry(1000, []);
 
-        $this->assertSame(
-            [
-                'file-abcd-1234.txt' => 1541410003,
-                'file_defg-4567.bin' => 1541410005,
-                'test_4444-4567.bin' => 1541410009,
-            ],
-            json_decode($outState, true),
-            'state.json was not saved correctly into out folder.'
-        );
+        return [
```

Comment:
```
    public function firstRunDataProvider(): array
    {
        $registry = $this->getRegistry(1000, []);

        return [
            ['dir1/files/1.txt', 900, $registry, false],
            ['dir1/files/2.txt', 1000, $registry, true],
            ['dir1/files/3.txt', 1002, $registry, true],
            ['dir1/files/4.txt', 1005, $registry, true],
            ['dir1/files/4.txt', 1001, $registry, true], <---------------- 1001 je novější než 1000!
        ];
    }


root@498d69f3665f:/code# composer tests-phpunit                                                                                                                                                  
> phpunit                                                                                                                                                                                        
PHPUnit 7.4.3 by Sebastian Bergmann and contributors.                                                                                                                                            
                                                                                                                                                                                                 
..............F......                                             21 / 21 (100%)                                                                                                                 
                                                                                                                                                                                                 
Time: 4.41 seconds, Memory: 6.00MB                                                                                                                                                               
                                                                                                                                                                                                 
There was 1 failure:                                                                                                                                                                             
                                                                                                                                                                                                 
1) Keboola\FtpExtractor\Tests\FileStateRegistryTest::testRegistry with data set #11 ('dir1/files/4.txt', 1001, Keboola\FtpExtractor\FileStateRegistry Object (...), true)                        
Failed asserting that false is identical to true.                                                                                                                                                
                                                                                                                                                                                                 
/code/tests/phpunit/FileStateRegistryTest.php:18                                                                                                                                                 
                                                                                                                                                                                                 
FAILURES!                                                                                                                                                                                        
Tests: 21, Assertions: 21, Failures: 1.                                                                                                                                                          
Script phpunit handling the tests-phpunit event returned with error code 1                                                                                                                       
```

Reply:
Tohle issue si uvědomuju. Souvisí to s tím, že jsem to předělal na styl jaký použil najlos v https://github.com/keboola/aws-s3-extractor/blob/ff8687b5715a24e25dd6c0647f42ad1a79960f61/src/Keboola/S3Extractor/Extractor.php#L228

Nevýhodá toho je, že to pak musí dostávat seřazené pole souborů podle timestamp. To jsem doplnil tady https://github.com/keboola/ex-ftp/pull/1/files/b09d46a0095bd97cd45703eeb0b12f3007d93011..569565db938cc4046deee30fff741b850b0e3d0f#diff-e44ef92d24ee0a2f3aca181e03ed76ecR100

A ty to pole chceš mít seřezené, protože při prvním běhu neexistuje žádný poslední timestamp. Pokud by jako první soubor přišel file s `timestamp 10` a jako další s `timestamp 2` tak ten s `timestamp 2` se nikdy nestáhne.

Reply:
> Pokud by jako první soubor přišel file s timestamp 10 a jako další s timestamp 2 tak ten s timestamp 2 se nikdy nestáhne.

To mi na tom algoritmu přijde dost nebezpečné. A mám pocit, že musí existovat jasnější algoritmus, který je snadější "to reason about".